File: src/main/scala/com/kirkwoodwest/scalawoods/GridButtonState.scala
package com.kirkwoodwest.scalawoods

enum GridButtonState(val value: Int):
  case Off extends GridButtonState(0)
  case HasClip extends GridButtonState(103)
  case ClipPlaying extends GridButtonState(43)
  case ClipPlayingNote extends GridButtonState(41)
  case NoteRecording extends GridButtonState(60)
  case NoteOffRecording extends GridButtonState(70)

File: src/main/java/com/kirkwoodwest/launchpadminimk3/hardware/GridButtonState.java
package com.kirkwoodwest.launchpadminimk3.hardware;

public enum GridButtonState {
    Off(0),
    HasClip(103),
    ClipPlaying(43),
    ClipPlayingNote(41),
    NoteRecording(60),
    NoteOffRecording(70);

    private final int value;

    GridButtonState(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}
